### PR TITLE
set old licenses to inactive and add refs to new licenses

### LINKED ISSUE #1135: Add 4.0 CC licenses and other software licenses
### Descriptive summary

This updates the `licenses.yml` file with more up-to-date licenses.

### Related work

https://github.com/psu-stewardship/scholarsphere/commit/4e0af852c

### PR DESCRIPTION
**Pull request by csyversen**
Fixes #1135  

We set the old licenses to inactive so you can still pull up works that use an old license, but they aren't selectable for new works.  

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by csyversen** [comment]
this keeps failing on one batch edit test, which apparently are known to be a bit flaky. I can run the hyrax tests locally and everything passes though. @samvera/hyrax-code-reviewers

**Comment by cpd3149** [comment]
hi @wickr i agree with your suggestions about the creative commons licenses, but are you saying the 'In Copyright' rights statement should be a user option for the rights field?

i thought that's what you suggested here: https://github.com/samvera/hyrax/issues/589

cc: @csyversen

**Comment by mjgiarlo** [comment]
I believe @wickr is suggesting keeping rights statements and licenses separate. :+1: to that, since the URI for "In Copyright" is already in the rights statements authority.

**Comment by csyversen** [comment]
Sound good, i just updated the licenses file to take the suggestions into account. Good catch on the duplicate entries, @wickr !

File: lib/generators/hyrax/templates/config/authorities/licenses.yml [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
+    - id: http://www.europeana.eu/portal/rights/rr-r.html
+      term: All rights reserved
+      active: false
+    - id: https://creativecommons.org/licenses/by/4.0/
+      term: Creative Commons BY Attribution 4.0 International
       active: true
-    - id: http://creativecommons.org/publicdomain/mark/1.0/
-      term: Public Domain Mark 1.0
+    - id: https://creativecommons.org/licenses/by-sa/4.0/
+      term: Creative Commons SA ShareAlike 4.0 International
```
Comment (mjgiarlo) [inline lib/generators/hyrax/templates/config/authorities/licenses.yml]: Should this be BY-SA to match the changes you made below?

File: lib/generators/hyrax/templates/config/authorities/licenses.yml [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
+      active: true
+    - id: https://creativecommons.org/licenses/by-nd/4.0/
+      term: Creative Commons ND NoDerivatives 4.0 International
```
Comment (mjgiarlo) [inline lib/generators/hyrax/templates/config/authorities/licenses.yml]: Should this be BY-ND to match the changes you made below?

File: lib/generators/hyrax/templates/config/authorities/licenses.yml [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+      active: true
+    - id: https://creativecommons.org/licenses/by-nc/4.0/
+      term: Creative Commons NC NonCommercial 4.0 International
```
Comment (mjgiarlo) [inline lib/generators/hyrax/templates/config/authorities/licenses.yml]: Should this be BY-NC to match the changes you made below?

File: lib/generators/hyrax/templates/config/authorities/licenses.yml [hunk 4]
```diff
@@ -1,28 +1,46 @@
...
+      active: true
+    - id: https://creativecommons.org/licenses/by-nc-nd/4.0/
+      term: Creative Commons BY-NC-ND NonCommercial-NoDerivs 4.0 International
+      active: true
+    - id: https://creativecommons.org/licenses/by-nc-sa/4.0/
+      term: Creative Commons BY-NC-SA NonCommercial-ShareAlike 4.0 International
       active: true
     - id: http://creativecommons.org/publicdomain/zero/1.0/
-      term: CC0 1.0 Universal
+      term: Creative Commons0 1.0 Universal
```
Comment (mjgiarlo) [inline lib/generators/hyrax/templates/config/authorities/licenses.yml]: This license is commonly known as "CC0", so I might replace this text with:

```
Creative Common CC0 1.0 Universal
```

**Comment by mjgiarlo** [comment]
@csyversen I have a few more cosmetic changes to request for consistency (which I hadn't noticed before). Then I think this is ready to go.

**Comment by csyversen** [comment]
ok now it's beautiful! @mjgiarlo

**Comment by mjgiarlo** [comment]
@csyversen Thanks for your patience. Feel free to merge when the PR goes green! And if it turns red because of silly feature specs, lmk and I can intervene.

**Comment by wickr** [comment]
and since this includes the change in PR #1598  @mjgiarlo I'd be fine if you close that one, or I can.
which makes this PR close #589 too

**Comment by csyversen** [comment]
@wickr gah, updating this real fast!

**Comment by csyversen** [comment]
@mjgiarlo sorry I had to update the term for the CC0 entry. There were tests hard coded to look for specific text so i'm guessing that was chosen for a reason. The only test failing now is the weird batch one that's known to be wonky. 

And this PR has a few commits where things are just going back and forth, want me to open up a new PR with squashed commits?

**Comment by mjgiarlo** [comment]
@csyversen I'll toss a commit up that restores the text you had before and fixes the tests. Give me a little bit, and let's look to get this merged on Tuesday. Cool?

**Comment by csyversen** [comment]
sounds good!

**Comment by mjgiarlo** [comment]
@csyversen OK, pushed the changes. I will check back on this on Tuesday and squash it at point of merger, so no worries there. Thanks for yr patience!

**Comment by mjgiarlo** [comment]
@csyversen Thanks again for taking this on.

**Comment by csyversen** [comment]
no problem! I feel like you did most of the work, so thanks for cleaning all this up!

**Pull request merged**